client/components/editor/teaching-elements/Assessment/TextSelection/Selections.js

```diff
@@ -0,0 +1,43 @@
+import cloneDeep from 'lodash/cloneDeep';
+import curry from 'lodash/curry';
+import curryRight from 'lodash/curryRight';
+import first from 'lodash/first';
+import inRange from 'lodash/inRange';
+import last from 'lodash/last';
+import reject from 'lodash/reject';
+
+export default class Selections {
+  constructor(selection = []) {
+    this.ranges = cloneDeep(selection);
```

Comment:
Hmm, if selection is array of ranges and every range is actually a pair of start and end index then it is not required to use deep clone here? 🤔 
It is sufficient to simply copy by slicing? 🤔